packages/framework/aqueduct/api-report/aqueduct.legacy.alpha.api.md

```diff
@@ -99,6 +105,7 @@ export abstract class PureDataObject<I extends DataObjectTypes = DataObjectTypes
     protected initializingFromExisting(): Promise<void>;
     // (undocumented)
     protected initProps?: I["InitialState"];
+    static readonly policies?: Partial<IFluidDataStorePolicies>;
```

Comment:
i'm not sure how i feel about this pattern, specifically, having the policies as a static on the data object itself. what i like is the co-locality of the policies with the thing the user writes, the data object. The other option is to pass it thru the factory, which is a bit decoupled from the data object itself, but also inline with how we handle other parts of the data object, like the sharedObjectRegistry.

Reply:
Yeah it took me some time to grasp how it works this way. Are there enough degrees of freedom using the static? It means it's fixed for a given derived class declaration. Maybe that's fine 🤷‍♂️ but it would mean you can't reuse a derived class and get different behaviors based on parameters or something (which you could if you passed it through the factory, if I'm understanding correctly).

It's probably fine, assuming it will work for Loop's use cases. I haven't spent enough time reading their DataObjects. This approach is nice and concise at least.

Reply:
derived types should still be able to override the static

Reply:
here is what it looks like to take through the factory: https://github.com/anthony-murphy/FluidFramework-1/compare/sm/readonlyInStagingMode...anthony-murphy:FluidFramework-1:sm/readonlyInStagingMode-pf?

the real downside is the number of args for the data object factories, so passing the policy leads to having to deal with a bunch of default value args. really it just needs to be moved to a props object.